### ISSUE TITLE
Using 'constructor' keyword within input breaks the training

### ISSUE THREAD

**Issue by didkivskyy** [BrainJS/brain.js#681]
![That's not a word](https://media.giphy.com/media/3CfAKeVE6CTQs/giphy.gif)
## _What_ is wrong?
Training iterations produce **NaN** in place of training error value. Happens when _constructor_ keyword is used within the input.

## _Where_ does it happen?
It's reproducible both in Node.js and Chrome browser (NeuralNetwork or NeuralNetworkGPU).

## _How_ do we replicate the issue?
Minimum reproducible node.js code:
```
const brain = require('brain.js');
const net = new brain.NeuralNetwork();
net.train (
     [
              { "input": { "anything": 1 },     "output": { "no_matter": 1 }},
              { "input": { "constructor": 1, }, "output": { "whatever": 1 }}
     ],
     { iterations: 5, log: true, logPeriod: 1 });
```

The root of the problem is the **constructor** keyword used as one of the input object properties. Removing it or just changing to anything similar like **constructor1** escapes the problem.
! **Please note**, the problem appears only with 2 or more input elements. I.e. if the first input element is removed and we try to train with _constructor_ input only, it gives no errors.

## Expected behavior (i.e. solution)
I guess this happens because the input interferes with one of the object properties inside **brain.js**. I see it being mentioned hundreds of times within the library.

Some sort of subtitution/escaping would help to eliminate this.

### Nodejs:
Nodejs v12.19.0 on Windows 10
### Browser:
Chrome Version 87.0.4280.88 (Official Build) (64-bit)
### Brain.js:
brain.js@2.0.0-beta.2
## _How_ important is this (1-5)?
3

## Other Comments
I know I can just remove this input for now.
But first of all **constructor** is one of my real inputs from my dataset.
And also this **NaN** error is not verbose at all and it takes a lot of time to spot the wrong input.

Many thanks!

**Comment by mubaidr** [comment]
Thanks for detailed issue filing. We will look into this issue.

**Comment by robertleeplummerjr** [comment]
Fixed in version 2.0.0-beta.14

Fix: https://github.com/BrainJS/brain.js/pull/783/files#diff-d9e7aef7a13113f8f6bf127abf13f29746931832a728bc6fa5ad83f4b02c3e9aR36
Unit test: https://github.com/BrainJS/brain.js/pull/783/files#diff-279643acb105095021a03450f3eb5f923fd8a31bd8bd997524e19fa73b9f5934R84

**Comment by robertleeplummerjr** [comment]
Working example: https://observablehq.com/@robertleeplummerjr/brain-js-issue-681
